File: New folder/rsII_exam_template_2025_26-main/eCommerce/eCommerce.Services/Database/ProductReview.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCommerce.Services.Database
{
    public class ProductReview
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [Range(1, 5)]
        public int Rating { get; set; }
        
        [MaxLength(1000)]
        public string Comment { get; set; } = string.Empty;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // User who wrote the review
        public int UserId { get; set; }
        
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
        
        // Product that was reviewed
        public int ProductId { get; set; }
        
        [ForeignKey("ProductId")]
        public Product Product { get; set; } = null!;

        /// <summary>Optional link to the order that contained this product (used to verify purchase).</summary>
        public int? OrderId { get; set; }

        [ForeignKey("OrderId")]
        public Order? Order { get; set; }
        
        // Whether the review is approved/visible
        public bool IsApproved { get; set; } = false;
    }
} 

File: Workshops/24-06-26 - Cards/eCommerce/UI/ecommerce_desktop/lib/models/product_review.dart
import 'package:json_annotation/json_annotation.dart';

part 'product_review.g.dart';

@JsonSerializable()
class ProductReview {
  final int id;
  final int rating;
  final String comment;
  final DateTime createdAt;
  final int userId;
  final String reviewerDisplayName;
  final int? orderId;
  final int? productId;


  ProductReview({
    required this.id,
    required this.rating,
    required this.comment,
    required this.createdAt,
    required this.userId,
    required this.reviewerDisplayName,
    this.orderId,
    this.productId
  });

  factory ProductReview.fromJson(Map<String, dynamic> json) =>
      _$ProductReviewFromJson(json);

  Map<String, dynamic> toJson() => _$ProductReviewToJson(this);
}
